newIDE/app/src/LayersList/index.js

```diff
@@ -46,69 +46,73 @@ class LayersListBody extends Component<*, LayersListBodyState> {
+            key={'layer-' + layerName}
+            layer={layer}
+            layerName={layerName}
+            nameError={this.state.nameErrors[layerName]}
+            effectsCount={layer.getEffectsCount()}
+            onEditEffects={() => onEditEffects(layer)}
+            isLightingLayer={isLightingLayer}
+            openLightingLayerDialog={() =>
+              this.setState({
+                ...this.state,
+                isLightingLayerDialogOpen: true,
+              })
             }
+            onBlur={event => {
+              const newName = event.target.value;
+              if (layerName === newName) return;
+
+              let success = true;
+              if (layersContainer.hasLayerNamed(newName)) {
+                success = false;
+              } else {
+                this.props.onRenameLayer(layerName, newName, doRename => {
+                  if (doRename)
+                    layersContainer.getLayer(layerName).setName(newName);
+                });
+              }
+
+              this.setState({
+                nameErrors: {
+                  ...this.state.nameErrors,
+                  [layerName]: !success,
+                },
+              });
+            }}
+            onRemove={() => {
+              this.props.onRemoveLayer(layerName, doRemove => {
+                if (!doRemove) return;
 
-            this.setState({
-              nameErrors: {
-                ...this.state.nameErrors,
-                [layerName]: !success,
-              },
-            });
-          }}
-          onRemove={() => {
-            this.props.onRemoveLayer(layerName, doRemove => {
-              if (!doRemove) return;
-
-              layersContainer.removeLayer(layerName);
+                layersContainer.removeLayer(layerName);
+                this._onLayerModified();
+              });
+            }}
+            isVisible={layer.getVisibility()}
+            onChangeVisibility={visible => {
+              layer.setVisibility(visible);
               this._onLayerModified();
-            });
-          }}
-          isVisible={layer.getVisibility()}
-          onChangeVisibility={visible => {
-            layer.setVisibility(visible);
-            this._onLayerModified();
-          }}
-          width={width}
-        />
-        <LightingLayerDialog
-          layer={layer}
-          open={this.state.isLightingLayerDialogOpen}
-          closeLightingLayerDialog={() => this.setState({
-            ...this.state,
-            isLightingLayerDialogOpen: false,
-          })}
-        />
+            }}
+            width={width}
+          />
+          {isLightingLayer && (
+            <LightingLayerDialog
+              layer={layer}
+              open={this.state.isLightingLayerDialogOpen}
+              closeLightingLayerDialog={() =>
+                this.setState({
+                  ...this.state,
+                  isLightingLayerDialogOpen: false,
+                })
```

Comment:
nit: `this.setState({ isLightiningLayerDialogOpen: false })`

setState is already doing shallow merge